internal/testutil/asserter/assert.go

```diff
@@ -175,32 +150,44 @@ func entitiesFor(entities []*integration.Entity, pseudotype string) []*integrati
 	return appropriateEntities
 }
 
-// entityHas returns true if supplied entity has metric m with type _similar_ to mType, false otherwise.
-func entityHas(e *integration.Entity, m string, mType metric.SourceType) bool {
+// entityMetric is a helper function that returns the first metric from an entity that matches the given name.
+func entityMetric(e *integration.Entity, m string) interface{} {
+	for _, ms := range e.Metrics {
+		entityMetric, found := ms.Metrics[m]
+		if found {
+			return entityMetric
+		}
+	}
+
+	return nil
+}
+
+// EntityHas returns true if the specified entity has a metric of name m equal to value.
+func EntityHas(e *integration.Entity, m string, value interface{}) bool {
 	// Wildcard metrics are ignored.
 	// TODO: Improve this and check matching glob patterns.
 	if strings.HasSuffix(m, "*") {
 		return true
 	}
 
-	for _, ms := range e.Metrics {
-		entityMetric, found := ms.Metrics[m]
-		if !found {
-			continue
-		}
-
-		// Check if metricType is an attribute but metric is not a string
-		_, isString := entityMetric.(string)
-		if isString && mType != metric.ATTRIBUTE {
-			continue
-		}
-
-		if !isString && mType == metric.ATTRIBUTE {
-			continue
-		}
+	return entityMetric(e, m) == value
+}
 
+// EntityContains returns true if supplied entity has metric m with type _similar_ to mType, false otherwise.
+func EntityContains(e *integration.Entity, m string, mType metric.SourceType) bool {
```

Comment:
I do not like particularly this name, EntityContains since it is actually checking if the mType ~matches the value type

Reply:
I'm not very convinced either about the value of those two funcs 🤔 

WDYT about `EntityHasValue` and `EntityHasType`?

Reply:
A better idea: `EntityMetricIs` and `EntityMetricKindIs`

Reply:
I like them all no bit preference 😄

Reply:
I'll go for the last one then :)